cpp/src/arrow/engine/substrait/relation_internal.cc

```diff
@@ -52,159 +55,239 @@ Status CheckRelCommon(const RelMessage& rel) {
   return Status::OK();
 }
 
-Result<DeclarationInfo> FromProto(const substrait::Rel& rel, const ExtensionSet& ext_set,
-                                  const ConversionOptions& conversion_options) {
-  static bool dataset_init = false;
-  if (!dataset_init) {
-    dataset_init = true;
-    dataset::internal::Initialize();
+Status CheckReadRelation(const substrait::ReadRel& rel,
```

Comment:
doubt: at a certain point, these unimplemented once will be implemented and the use of this function wouldn't be very useful. Wouldn't it be better to leave them within the main function where we do the rest of the tasks?

Reply:
I did this so that the structure would be consistent and thus easier to read.

think that at some point when we don't need it anymore we can just drop it. Since it is only internal to this source file, there shouldn't be any problem with the structure changing (I think).

Reply:
I removed this as part of backing out the re-organization. I will include it in a subsequent PR.